scipy/optimize/optimize.py

```diff
@@ -2102,7 +2105,10 @@ def _minimize_scalar_golden(func, brack=None, args=(),
     f1 = func(*((x1,) + args))
     f2 = func(*((x2,) + args))
     funcalls += 2
-    while (numpy.abs(x3 - x0) > tol * (numpy.abs(x1) + numpy.abs(x2))):
+    nit = 0
+    for i in xrange(maxiter):
```

Comment:
I don't understand why the interface should be different compare to the other function. Isn't it better to make the same interface for functions? I've done it in that way because that how it works in the ``brent`` function.

In addition ``funcalls`` is a non-zero value when first iteration starts. So it's possible that during the first iteration ``funcalls`` is higher than ``maxiter``. For instance, ``brent`` function makes 20 class and 2 more for initialization step, which means that we have 22 function calls. If user set up ``maxiter = 10`` that would be odd to see that algorithm needs 12 more funcalls before it's reach the limit condition.